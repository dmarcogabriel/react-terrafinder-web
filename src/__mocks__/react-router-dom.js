const originalModule = jest.requireActual('react-router-dom');

module.exports = {
  ...originalModule,
  useHistory: () => ({
    history: { push: jest.fn() },
  }),
};
