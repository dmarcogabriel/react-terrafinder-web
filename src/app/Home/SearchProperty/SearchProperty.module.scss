@import 'styles';

.container {
  display: flex;
}
