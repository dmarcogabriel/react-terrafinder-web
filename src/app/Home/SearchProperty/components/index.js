import CheckBox from './CheckBox';
import PropertiesList from './PropertiesList';

export { CheckBox, PropertiesList };
