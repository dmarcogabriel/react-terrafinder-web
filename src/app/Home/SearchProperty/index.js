import React, { useEffect, useState } from 'react';
import { Button } from 'common/components';
import { MdSearch } from 'react-icons/md';
import Property from 'common/components/Property';
import { useHistory } from 'react-router-dom';
import api from 'services/api';
import {
  container,
  card,
  searchButton,
  buttonLinkKind,
  title,
  cards,
  filtersCard,
  propertiesList,
} from './SearchProperty.module.scss';
import { CheckBox } from './components';
import propertyImg from './static/soja.jpg';

export default function SearchProperty() {
  const history = useHistory();
  const [properties, setProperties] = useState([]);

  const selectProperty = (id) => {
    history.push(`/home/property/${id}`);
  };

  const loadProperties = async () => {
    const res = await api.get('properties');

    setProperties(res.data.properties);
  };

  useEffect(() => {
    loadProperties();
  }, []);

  return (
    <div className={container}>
      <h1>Use os filtros abaixo para melhorar o resultado de sua busca</h1>

      <div className={filtersCard}>
        {/*
         // todo: Search Inputs
        */}

        <button type="button" className={buttonLinkKind}>
          Limpar Filtro
        </button>

        {/* <div className={cards}>
          <div className={card}>
            <p className={title}>Casa & Moradia</p>

            <CheckBox label="Número de Quartos" />
            <CheckBox label="Número de Banheiros" />
          </div>

          <div className={card}>
            <p className={title}>Tipos de solo e cultivo</p>

            <CheckBox label="Soja" />
            <CheckBox label="Milho" />
          </div>

          <div className={card}>
            <p className={title}>Atividades & diferenciais</p>

            <CheckBox label="Rio" />
            <CheckBox label="Lago" />
            <CheckBox label="Cachoeiras" />
            <CheckBox label="Animais Silvestres" />
          </div>
        </div> */}

        <Button className={searchButton}>
          <MdSearch size={22} />
          Buscar
        </Button>
      </div>
      {/* 
        //todo: add loading here
      */}
      <div className={propertiesList}>
        {properties.map((property, i) => (
          <Property
            i={i}
            key={property._id}
            property={property}
            onSelect={selectProperty}
            photo={propertyImg}
          />
        ))}
      </div>
    </div>
  );
}
