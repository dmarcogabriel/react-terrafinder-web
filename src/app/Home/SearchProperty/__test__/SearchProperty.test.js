import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
// import { render } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router-dom';
import SearchProperty from '..';

const history = createMemoryHistory();

const Comp = (props) => (
  <Router history={history}>
    <SearchProperty {...props} />
  </Router>
);

describe('<SearchProperty />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Comp />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  // Add tests here...

  it('matches snapshot', () => {
    const tree = renderer.create(<Comp />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
