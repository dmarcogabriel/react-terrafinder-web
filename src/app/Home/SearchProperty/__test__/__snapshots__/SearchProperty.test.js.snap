// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchProperty /> matches snapshot 1`] = `
<div>
  <h1>
    Use os filtros abaixo para melhorar o resultado de sua busca
  </h1>
  <div>
    <div
      className="Filters__container"
    >
      <div>
        <button
          onClick={[Function]}
          type="button"
        >
          <p
            label="propertyKind"
          >
            Tipo de Propriedade
          </p>
          <div>
            <p>
              Selecione...
            </p>
            <svg
              fill="currentColor"
              height={22}
              size={22}
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 24 24"
              width={22}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
        </button>
      </div>
      <div>
        <button
          onClick={[Function]}
          type="button"
        >
          <p>
            Área do Imóvel
          </p>
          <div>
            0 até 1
          </div>
        </button>
      </div>
      <div>
        <button
          onClick={[Function]}
          type="button"
        >
          <p>
            Estado
          </p>
          <div>
            <p>
              Selecione...
            </p>
            <svg
              fill="currentColor"
              height={22}
              size={22}
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 24 24"
              width={22}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
        </button>
      </div>
      <div>
        <button
          onClick={[Function]}
          type="button"
        >
          <p>
            Intervalo de Preço
          </p>
          <div>
            0 até 1
          </div>
        </button>
      </div>
      <button
        className="button searchButton"
        onClick={[Function]}
        type="button"
      >
        Procurar Imóvel
      </button>
    </div>
    <button
      onClick={[Function]}
      type="button"
    >
      Limpar Filtro
    </button>
  </div>
  <div />
</div>
`;
